### PR TITLE
update: Load ATB param from the site

### PR DESCRIPTION
**Pull request by mrshu**
* Load the full ATB param from a class of the HTML element that can be
  found on the loaded duckduckgo.com site at install time.

Signed-off-by: mr.Shu <mr@shu.io>

### CODE REVIEW

**Comment by mrshu** [comment]
@bsstoner The best way of testing this seems to be the following:

1. Change the value of [`is_installed`](https://github.com/duckduckgo/firefox-zeroclickinfo/blob/master/lib/main.js#L114) to `false`.
2. Run the extension using `jpm run`
3. Open https://mrshu.duckduckgo.com

If you do another search on DuckDuckGo.com, you should get the "full" ATB parameter.

File: data/js/duckduckgo.js [hunk 1]
```diff
@@ -1,4 +1,16 @@
-self.port.emit('loaded');
+// grab the ATB param from the html element's class
+function getATBParam() {
```
Comment (mrshu) [inline data/js/duckduckgo.js]: @bsstoner I am not sure if this is the best way of doing this in vanilla JavaScript but it is certainly the cleanest one I found.

File: data/js/duckduckgo.js [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+  for (var i = 0; i < htmlClasslist.length; i++) {
+    var cls = htmlClasslist[i];
+    if (cls.match(/^atb-/)) {
+      return cls.replace(/^atb-/, '');
+    }
+  }
+}
+
+var atbParam = getATBParam();
+self.port.emit('loaded', atbParam);
```
Comment (mrshu) [inline data/js/duckduckgo.js]: @bsstoner using it this way means that the `atbParam` might be `undefined` in some circumstances. Do you think it makes sense to check for that?

**Comment by mrshu** [comment]
@bsstoner the way of passing the ATB param to the extension has changed but my comment from above is still valid and the best way of actually testing this.

**Comment by bsstoner** [comment]
👍 @mrshu this looks good and the code makes sense to me. We're defaulting to the regular atb param, and only doing this when they install from our page, which should be fine.

I'm having a problem testing locally though, when I run `jpm run` I get an error:
```
1472695072312	addons.xpi	WARN	Add-on jid1-ZAdIEUB7XOzOJw@jetpack is not correctly signed.
```

It still opens the browser, but the extension is disabled and it won't let you enable it.

**Comment by bsstoner** [comment]
Nevermind, I figured it out. Apparently Firefox v48 disabled the ability to load unsigned extensions with no ability for user override :). I downgraded back to v47 and it works.

**Comment by mrshu** [comment]
@bsstoner that's just crazy, I normally use Firefox 48 without any problems when testing extensions. It might be that I have some settings overriden.

It seems that the only real way forward is to use Firefox Nightly (dev edition) going forward -- that should still have this enabled.

**Pull request merged**